.gitignore

```diff
@@ -0,0 +1,23 @@
+/.vs
```

Comment:
I'm not sure what you mean - you seem to have highlighted the .vs folder specifically; this is just the folder visual studio code holds cache files in and other stuff not related to building but editing. Or did you mean the whole .gitignore file?

Reply:
I mostly meant build artifacts you listed in .gitignore. I don't think those should be there.
Things like .vs and .vscode are probably ok actually.

Reply:
I like your idea of re-organizing the generated files, although i'm not convinced moving them outside of the checkout repository is a great idea (a small part of my heart burns in a fire when I see 'c:\' appear in any scripts) - keeping everything relative is a good thing, particularly within CI environments or containerized builds where you might not have reliable access to some out-of-checkout-folder areas.

However, I completely agree that the .gitignore I posted is a hot mess. But I would argue that while the default CMake configuration works the way it does, why not let it pass? At least until we do something with CMake to actually put things into a neat root folder that can be excluded...

I could certainly group all the 'nightmare mess' files under a comment header to group this for easier factoring out later if you want?

Reply:
How about adding something like .build to .gitignore and then 'mkdir -p .build && cd .build && cmake .. && make' ?

Reply:
OK, i just tried looking at this, and it's waaaay too big a thing to do and will just block this relatively modest PR for Win32 building. Trying your suggestion just lead to a big pile of knock-on problems locating libraries and such, and unpicking that is more than I can do any time soon.

I just reduced the scope of the .gitignore to block any VS Code project folders and any junk nuget might download (e.g. the pthreads package).

If a great CMake refactor is something people really still want, it's probably something best done by someone that does not have to hit the CMake man pages for every step like me :)

Reply:
OK, fair enough. It is strange that you run into anything that would require CMake changes: I imagined this would be as simple as adding a couple of lines to .appveyor.yml (mkdir .build && cd .build) and adjusting some paths in gather-package.bat but maybe I am missing something vital. I should try to do this on a branch myself and see what issues I run into. I did not mean to complicate things for you @lewk2 and thank you very much for your efforts!